### PR TITLE
Pin pulumi CLI

### PR DESCRIPTION
**Pull request by t0yv0**
Borrowing targets from pulumi-azure-native here to make sure code generation is pinned.

### CODE REVIEW

**Comment by t0yv0** [comment]
Guessing weekly-pulumi-update needs a small change too.

**Comment by t0yv0** [comment]
Oh this is not hermetic.

```
t0yv0@Antons-MacBook-Pro> make python_sdk
rm -rf sdk/python
.pulumi/bin/pulumi package gen-sdk --language python provider/cmd/pulumi-resource-command/schema.json
warning: using pulumi-language-python from $PATH at /Users/t0yv0/.nix-profile/bin/pulumi-language-python
error: method GeneratePackage not implemented
make: *** [python_sdk] Error 255

```

**Comment by t0yv0** [comment]
I've added a fix to make sure language plugins are hermetic and governed by .pulumi.version.

**Comment by t0yv0** [comment]
This is made hermetic now with follow-up commit. Thanks.

**Review by mjeffryes — APPROVED** [review]
nice

**Comment by t0yv0** [comment]
Filed https://github.com/pulumi/pulumi-command/issues/265 for later.

File: Makefile [hunk 1]
```diff
@@ -40,34 +43,34 @@ test_provider::
 	cd provider/pkg && go test -short -v -count=1 -cover -timeout 2h -parallel ${TESTPARALLELISM} ./...
 
 dotnet_sdk:: DOTNET_VERSION := $(shell pulumictl get version --language dotnet)
-dotnet_sdk::
+dotnet_sdk::	.pulumi/bin/pulumi
 	rm -rf sdk/dotnet
-	pulumi package gen-sdk --language dotnet $(SCHEMA_FILE)
+	.pulumi/bin/pulumi package gen-sdk --language dotnet $(SCHEMA_FILE)
```
Comment (iwahbe) [inline Makefile:48]: If we are adding this to `PATH`, we shouldn't need to prefix `pulumi`.

Comment (t0yv0) [reply Makefile:48]: See https://github.com/pulumi/pulumi-command/pull/266

**Pull request merged**